src/Editor.js

```diff
@@ -1234,6 +1257,97 @@ const Editor = () => {
                 Width{" "}&{" "}Height
               </Button>
             </div>
+            <div
+              style={{
+                display: "grid",
+                gridTemplateColumns: "repeat(4, min-content)",
+                alignItems: "center",
+                justifyItems: "center",
+                gap: 6,
+                padding: 6
+              }}
+            >
+              <Button
+                disabled={selection.size < 2}
+                onClick={() => {
+                  let widestSelectedLayer = null;
+                  doc.layers.forEach(layer => {
+                    if (
+                      selection.has(layer.id) &&
+                      (!widestSelectedLayer ||
+                        widestSelectedLayer.x2 - widestSelectedLayer.x1 <
+                          layer.x2 - layer.x1)
+                    ) {
+                      widestSelectedLayer = layer;
+                    }
+                  });
```

Comment:
We can refactor this using a [filter](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/filter) and [reduce](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/reduce) to be a little more methodical and functional:

```js
const allSelectedLayers = doc.layers.filter(layer => selection.has(layer.id));
const widestSelectedLayer = allSelectedLayers.reduce(
  (widestFoundSoFar, layer) => (
    widestFoundSoFar.x2 - widestFoundSoFar.x1 < layer.x2 - layer.x1
      ? layer
      : widestFoundSoFar
  ), // The return value of this function will be the value of the widestFoundSoFar argument when this function is called again in the next iteration
  { x1: 0, x2: 0 } // This is the initial value of widestFoundSoFar in the first iteration
);
```

Reply:
I also think it'd be good to move this into the `resizeLayerByExtreme` function so that we can use it something like this:

```js
resizeLayerByExtreme(layers, "widest", layer => selection.has(layer.id));
```

That way the `resizeLayerByExtreme` function can be responsible for both identifying the widest/narrowest/tallest/shortest layer as well as resizing the other layers.

Alternatively, this function could be called something like `setLayerDimensions` that we use like this:

```js
const layersSetToSameX = setLayerDimensions(layers, { x1: 100, x2: 200 });
const layersSetToSameY = setLayerDimensions(layers, { y1: 50, y2: 100 });
const layersSetToSameY2 = setLayerDimensions(layers, { y2: 400 });
```

The problem I'm trying to solve here is that `resizeLayerByExtreme` is a bit of a misleading name, because the `extreme` argument that is passed to it is really just deciding if it considers the `x` or `y` axis, but the function doesn't really do anything that considers extremes.

Reply:
This makes a lot of sense to me. I don't really get why the third argument is passed in as an argument or how it'll be used in the below snippet.
```
resizeLayerByExtreme(layers, "widest", layer => selection.has(layer.id));
```

Reply:
It makes it easier to use in this pattern (you'll notice a lot of the other functions in the `layers.js` file have similar signatures):

```js
setState(current => ({
  ...current,
  doc: {
    ...current.doc,
    layers: resizeLayerByExtreme(current.doc.layers, "widest", layer => selection.has(layer.id))
  }
}));
```

You can pass it an array of layers and a [predicate](https://en.wikipedia.org/wiki/Predicate_(mathematical_logic)) so that it can return an array of all the same layers (in the same order) with just the desired ones updated.

Reply:
Thanks for the explanations